### PR TITLE
tests: Add more GPU-AV tests

### PR DESCRIPTION
**Pull request by spencer-lunarg**
Extra tests to exercise end-to-end usage of the SPIR-V instrumentation

### CODE REVIEW

File: tests/unit/gpu_av.cpp [hunk 1]
```diff
@@ -809,6 +769,113 @@ TEST_F(NegativeGpuAV, UnnormalizedCoordinatesSeparateSamplerSharedSampler) {
     m_errorMonitor->VerifyFound();
 }
 
+TEST_F(NegativeGpuAV, ShareOpSampledImage) {
+    TEST_DESCRIPTION(
+        "have two OpImageSampleImplicitLod share the same OpSampledImage, which needs to be in the same block post-shader "
+        "instrumentation");
```
Comment (filnet) [inline tests/unit/gpu_av.cpp:775]: other descriptions end with a colon.

File: tests/unit/gpu_av_oob_positive.cpp [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+#include "../framework/layer_validation_tests.h"
+#include "../framework/pipeline_helper.h"
+#include "../framework/descriptor_helper.h"
+#include "../framework/gpu_av_helper.h"
+
+TEST_F(PositiveGpuAVOOB, ImageLoadStoreTexelFetch) {
+    TEST_DESCRIPTION("index into a image Load, Store, and texelFetch");
```
Comment (filnet) [inline tests/unit/gpu_av_oob_positive.cpp:20]: "into an image"

File: tests/unit/gpu_av_oob.cpp [hunk 3]
```diff
@@ -823,3 +812,480 @@ TEST_F(NegativeGpuAVOOB, GPLIndependentSets) {
         m_default_queue->wait();
     }
 }
+
+TEST_F(NegativeGpuAVOOB, StorageBuffer) {
+    TEST_DESCRIPTION("Make suree OOB is still checked when result is from a BufferDeviceAddress");
+    SetTargetApiVersion(VK_API_VERSION_1_2);
```
Comment (arno-lunarg) [inline tests/unit/gpu_av_oob.cpp:818]: `VK_API_VERSION_1_2` will be overwritten by `VK_API_VERSION_1_1` in `InitGpuAvFramework`

**Review by arno-lunarg — APPROVED** [review]
LGTM, one comment

File: tests/framework/shader_templates.h [hunk 4]
```diff
@@ -25,6 +25,19 @@ static const char kVertexMinimalGlsl[] = R"glsl(
...
+    #version 450
+    vec2 vertices[3];
+    void main(){
+        vertices[0] = vec2(-1.0, -1.0);
+        vertices[1] = vec2( 1.0, -1.0);
+        vertices[2] = vec2( 0.0,  1.0);
+        gl_Position = vec4(vertices[gl_VertexIndex % 3], 0.0, 1.0);
+    }
+)glsl";
+
```
Comment (jeremyg-lunarg) [inline tests/framework/shader_templates.h:40]: good call on pulling this shader up.

**Review by jeremyg-lunarg — APPROVED** [review]

File: tests/unit/gpu_av_oob.cpp (hunk 3 above)
Comment (spencer-lunarg) [reply tests/unit/gpu_av_oob.cpp:818]: no, the logic for `SetTargetApiVersion` checks and will only update if the new value passed in is higher

**Pull request merged**